src/implot_visualization/app/main.cpp

```diff
@@ -111,8 +112,35 @@ static void main_loop(void *arg) {
     ImGui_ImplSDL2_NewFrame();
     ImGui::NewFrame();
 
-    // Show counter demo
-    {
+    // Visualize One GR Signal
+    bool visualize_gr_signal = false;
+    if (visualize_gr_signal) {
+        fetch("http://localhost:8080/pulsed_power/timeDomainWorker/sinus", buffer);
+
+        ImGui::SetNextWindowSize(ImVec2(800, 400), ImGuiCond_Appearing);
+        ImGui::Begin("Sinus Demo Window");
+        if (ImPlot::BeginPlot("Sinus Sink")) {
+            static ImPlotAxisFlags xflags = ImPlotAxisFlags_None;
+            static ImPlotAxisFlags yflags = ImPlotAxisFlags_AutoFit | ImPlotAxisFlags_RangeFit;
+            ImPlot::SetupAxes("UTC Time", "Value", xflags, yflags);
+            auto   clock       = std::chrono::system_clock::now();
+            double currentTime = (std::chrono::duration_cast<std::chrono::milliseconds>(clock.time_since_epoch()).count()) / 1000.0;
+            ImPlot::SetupAxisLimits(ImAxis_X1, currentTime - 10.0, currentTime, ImGuiCond_Always);
+            ImPlot::SetupAxisScale(ImAxis_X1, ImPlotScale_Time);
+            ImPlot::SetNextFillStyle(IMPLOT_AUTO_COL, 0.5f);
+            if (buffer.Data.size() > 0) {
+                ImPlot::PlotLine("Sinus", &buffer.Data[0].x, &buffer.Data[0].y, buffer.Data.size(), 0, buffer.Offset, 2 * sizeof(double));
```

Comment:
Yes, I agree. It is already implemented in branch [visualize_multiple_gr_signals](https://github.com/fair-acc/pulsed-power-ml/tree/visualize_multiple_gr_signals), so it will be resolved with the corresponding PR.